packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -292,17 +294,24 @@ export function FileUploader({
         setFileStatuses(newFileStatuses);
       };
     },
-    [acceptedFileTypes, fileStatuses, setFileStatuses]
+    [fileStatuses, setFileStatuses]
   );
 
   const updateFileState = useCallback(
     (index: number, fileState: FileState) => {
       setFileStatuses((prevFileStatuses) => {
         const newFileStatuses = [...prevFileStatuses];
         const status = newFileStatuses[index];
+        // Check if extension is valid before setting state
+        const [extension] = status.name.split('.').reverse();
+        const [fileExtension] = status.file.name.split('.').reverse();
+        const validExtension = fileExtension === extension ? null : 'error';
```

Comment:
I would extract this into a util, it's also repeated on lines 282-283/4

Reply:
Extracted to `isValidExtension` util function.

Reply:
[f86e2e5](https://github.com/aws-amplify/amplify-ui/pull/3072/commits/f86e2e53201ee19a7098ee5d45abeb53faef2aa6)